### PR TITLE
chore: update github.com/kubernetes-sigs/application

### PR DESCRIPTION
**Pull request by pengx17**
Update sigs/application dependency.

https://github.com/kubernetes-sigs/application/pull/45 is merged and it has a breaking change on the Application List pr https://github.com/kubernetes/dashboard/pull/2980.

### CODE REVIEW

**Comment by pengx17** [comment]
@konryd looks like the dependencies of sigs/app has some compatibility issues with the current Dashboard dependencies. This pr should be put on-hold until the sig/apps team get the kube-builder updated in the pr https://github.com/kubernetes-sigs/application/pull/52.

File: vendor/github.com/kubernetes-sigs/application/pkg/apis/app/v1alpha1/application_types.go [hunk 1]
```diff
@@ -17,22 +17,15 @@ limitations under the License.
 package v1alpha1
 
 import (
+	core "k8s.io/api/core/v1"
```
[the remark this answers is not in the document]
Comment (konryd) [reply vendor/github.com/kubernetes-sigs/application/pkg/apis/app/v1alpha1/application_types.go]: Wait, you're saying that "dep ensure" pulled in the latest kubernetes-sigs/application code and then edited this one line for you? That's strange. @kow3ns, @mattfarina - do you recognize this behavior?

**Comment by konryd** [comment]
/approve

I'm not sure about the "dep ensure" weirdness, but I'd rather not block on it. If the application builds and runs, we can resolve it separately.

**Comment by pengx17** [comment]
@konryd Sorry I mislead you. What I mean is that after running `dep ensure`, the `sigs/application` package introduces an import from k8s.io/kubernetes/pkg/apis/core` package in the file ( `k8s.io/kubernetes/pkg/apis/core`). However the new included file (`k8s.io/kubernetes/pkg/apis/core`) is missing in the vendors. I am not so sure why this happened thus what I did here was to replace it with `k8s.io/api/core/v1` to make sure Dashboard builds again.

After I saw the pr (kubernetes-sigs/application#52) where it did some dependency updates work, I realized that the missing dependencies may be caused by the fact that there were some dependency mismatches between Dashboard and the `sigs/app` client. I guess the current master branch was built with a newer version of kubebuilder but the maintainer did not update `Gopkg.toml` accordingly.

**Pull request merged**